src/coreclr/.nuget/ILCompiler.Reflection.ReadyToRun/ILCompiler.Reflection.ReadyToRun.pkgproj

```diff
@@ -0,0 +1,27 @@
+<Project DefaultTargets="Build">
+  <Import Project="$([MSBuild]::GetPathOfFileAbove(Directory.Build.props))" />
+
+  <PropertyGroup>
+    <SkipValidatePackage>true</SkipValidatePackage>
+    <PackageDescription>ILCompiler.Reflection.ReadyToRun.dll</PackageDescription>
+    <PreReleaseVersionLabel>alpha</PreReleaseVersionLabel>
+    <PreReleaseVersionIteration>1</PreReleaseVersionIteration>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <ProjectReference Include="..\..\tools\aot\ILCompiler.Reflection.ReadyToRun\ILCompiler.Reflection.ReadyToRun.csproj" />
+  </ItemGroup>
+
+  <ItemGroup>
+    <PackageFile Include="$(RuntimeBinDir)R2RDump\ILCompiler.Reflection.ReadyToRun.dll">
+       <TargetPath>\lib\netstandard2.0\</TargetPath>
+    </PackageFile>
+    <Dependency Include="System.Reflection.Metadata">
+       <Version>$(SystemReflectionMetadataVersion)</Version>
+       <TargetFramework>netstandard2.0</TargetFramework>
+       <Exclude>Build,Analyzers</Exclude>
+    </Dependency>
```

Comment:
@ericstj is it ok to depend on a LKG here?

Reply:
1. Note that the binary itself is built with the same version: https://github.com/dotnet/runtime/blob/c5d16e0f184cb6e759dcb7e03b155c63886d88de/src/coreclr/tools/aot/ILCompiler.Reflection.ReadyToRun/ILCompiler.Reflection.ReadyToRun.csproj#L21
2. ILSpy is doing something interesting in their build:
https://github.com/icsharpcode/ILSpy/blob/7211587b456765126dcf3556bdd612fdffe9d76c/ILSpy.ReadyToRun/ILSpy.ReadyToRun.csproj#L40-L50

3. Their `SystemReflectionMetadataVersion` happens to be the same with ours:

https://github.com/dotnet/runtime/blob/6ca0471736281ac91a5bc86354135e204df47a11/eng/Versions.props#L90

https://github.com/icsharpcode/ILSpy/blob/dfd82de1afdb7d897d1c0084856a84197ac8fd20/packages.props#L9

Reply:
Why wouldn't we build against the live System.Reflection.MetaData?

Reply:
It appears to be used consistently across the crossgen2 compiler, for example, in the type system:

https://github.com/dotnet/runtime/blob/ee0719079c4c4540f5adb135f002ad3ecf6c25ea/src/coreclr/tools/aot/ILCompiler.TypeSystem.ReadyToRun/ILCompiler.TypeSystem.ReadyToRun.csproj#L30-L32

But I don't really know why, @dotnet/crossgen-contrib to comment on it.

That being said, ILSpy is currently using 5.0.0, if we publish a version that depends on a higher version of `System.Reflection.Metadata`, they might not be able to consume it, defeating the purpose of this packaging work.

FYI, @christophwille

Reply:
Our package https://www.nuget.org/packages/ICSharpCode.Decompiler/7.0.0.6472-rc1 must target SRM version from https://www.nuget.org/packages/Microsoft.CodeAnalysis.Common/. The UI could target newer, but... if the newer SRM version has a breaking API change or behavior change this then means we can't use it.

Reply:
At this very point in time, it would be ok (@siegfriedpammer just checked): https://www.fuget.org/packages/System.Reflection.Metadata/6.0.0-preview.3.21201.4/lib/netstandard2.0/diff/5.0.0/ - but that isn't guaranteed for the future. In the end, keeping in lockstep with Roslyn is more important than our UI - so we could end up in a situation where we'd like a newer R2R version, but can't update because ics.d's SRM must take precedence in our considerations.

Reply:
https://github.com/dotnet/roslyn/issues/50832 might be related to this.

Reply:
Any specific concerns about depending on LKG? Agree that we should ideally depend on live versions, which we could do when we get close to shipping .net 6.

Reply:
No concerns from my side except increased complexity in the build graph and sourcebuild. But we already depend on LKGs in other places, so it's ok to introduce yet another LKG dependency.

Reply:
ok sounds good. We can certainly re-evaluate the dependency once we are close to wrapping up 6. @cshung perhaps we should create a tracking item to revisit this later.